packages/dumb/src/components/stats/insulin/insulin-stat.tsx

```diff
@@ -0,0 +1,205 @@
+import { useLocation } from 'react-router-dom'
+import { EMPTY_DATA_PLACEHOLDER } from '../../../models/stats.model'
+import { useTheme } from '@mui/material/styles'
+
+interface TotalInsulinPropsData {
+  id: string
+  title: string
+  units: string
+  value: number
+  valueString: string
+}
+
+export interface TotalInsulinStatProps {
+  data: TotalInsulinPropsData[]
+  totalInsulin: number
+  weight: number | string
+  dailyDose: number
+}
+
+const InsulinStat: FunctionComponent<TotalInsulinStatProps> = (props) => {
+  const {
+    data,
+    totalInsulin,
+    dailyDose,
+    weight
+  } = props
+  const theme = useTheme()
+  const location = useLocation()
+  const isDailyPage = location.pathname.includes('daily')
+  const isDashboardPage = location.pathname.includes('dashboard')
+  const isTrendsPage = location.pathname.includes('trends')
+  const annotations = [t(isDailyPage ? 'total-insulin-days-tooltip' : 'average-daily-insulin-tooltip'), t('total-insulin-how-calculate-tooltip')]
+  const isDisabledWeight = weight === EMPTY_DATA_PLACEHOLDER
+  if (data.length === 0) {
+    annotations.push(t('tooltip-empty-stat'))
+  }
+  const dailyDosePerWeight = (): string | number => {
+    if (weight === EMPTY_DATA_PLACEHOLDER) {
+      return EMPTY_DATA_PLACEHOLDER
+    }
+    const value = dailyDose / +weight
+    return value > 0 && Number.isFinite(value) ? formatDecimalNumber(value, 2) : EMPTY_DATA_PLACEHOLDER
+  }
+  const isDailyDosePerWeight = dailyDosePerWeight.toString() === EMPTY_DATA_PLACEHOLDER
+
+  const getOutPutValueClasses = (): string => {
+    return `${isDailyDosePerWeight ? `${styles.outputValueDisabled}` : `${styles.dailyDoseValue}`}`
+  }
+
+  const getPercentage = (value: number): string => {
+    const res = Math.round(100 * value / totalInsulin)
+    return res > 0 ? res.toString(10) : EMPTY_DATA_PLACEHOLDER
+  }
+
+  return (
+    <div data-testid="container-insulin-stats">
+      <Box>
+        <Box
+          display="flex"
+          justifyContent="space-between"
+          alignItems="center"
+          marginBottom="4px"
+        >
+          <Box>
+            {isDailyPage ? t('total-insulin') : t('average-daily-total-insulin')}
+            <StatTooltip
+              annotations={annotations}
+            />
+          </Box>
+          <Box
+            display="flex"
+            justifyContent="space-between"
+            alignItems="baseline"
+          >
+            <span className={styles.titleTotal}>
+              {totalInsulin}
+            </span>
+            <span className={styles.titleSuffix}>
+              U
+            </span>
+          </Box>
+        </Box>
+        {data.map(entry => {
+          return (
+            <React.Fragment key={entry.id}>
+              <Box
+                display="flex"
+                justifyContent="space-between"
+                alignItems="center"
+                margin="0px 0px 4px 8px"
```

Comment:
Same here useless margin bottom, and for the left margin you should use `theme.spacing(1)`